libminifi/src/core/ProcessSession.cpp

```diff
@@ -965,6 +965,18 @@ std::shared_ptr<core::FlowFile> ProcessSession::get() {
   return NULL;
 }
 
+bool ProcessSession::outgoingConnectionsFull(const std::string& relationship) {
+  std::set<std::shared_ptr<Connectable>> connections = process_context_->getProcessorNode()->getOutGoingConnections(relationship);
+  Connection * connection = nullptr;
+  for (const auto& conn : connections) {
+    connection = dynamic_cast<Connection*>(conn.get());
+    if (connection && connection->isFull()) {
+      return true;
+    }
+  }
+  return false;
```

Comment:
I prefer to use STL algorithms when possible.
```
return std::find_if(std::cbegin(connections), std::cend(connections), [](const auto& conn) {
    auto* connection = dynamic_cast<Connection*>(conn.get());
    return connection && connection->isFull();
}) != std::cend(connections);
```

Reply:
I also like std algorithms, but it in this I don't think it makes the code any more readable.

Reply:
Maybe `std::any_of` would be better, I've just learned about it. I still prefer anything higher level than a raw loop.
Also see: C++ seasoning at ~4:30 https://www.youtube.com/watch?v=qH6sSOr-yk8

I've already approved the PR since this is a minor detail, so feel free to merge it either way.